local-server/local-server-services/src/main/java/com/esri/samples/localserver/LocalServerServicesController.java

```diff
@@ -0,0 +1,287 @@
+            serviceUrl = Paths.get(getClass().getResource("/PointsofInterest.mpk").toURI()).toString();
+            localService = new LocalMapService(serviceUrl);
+            break;
+          case "Feature Service":
+            serviceUrl = Paths.get(getClass().getResource("/PointsofInterest.mpk").toURI()).toString();
+            localService = new LocalFeatureService(serviceUrl);
+            break;
+          case "Geoprocessing Service":
+            serviceUrl = Paths.get(getClass().getResource("/MessageInABottle.gpk").toURI()).toString();
+            localService = new LocalGeoprocessingService(serviceUrl);
+        }
+
+        startLocalService(localService, serviceSelected);
+
+      } else {
+        // if service is currenlt running on server warn user
+        Platform.runLater(() -> {
+          Alert dialog = new Alert(AlertType.INFORMATION);
+          dialog.setHeaderText("Service Running");
+          dialog.setContentText(serviceSelected + " has already been started.");
+          dialog.showAndWait();
+        });
+      }
+    }
+  }
+
+  /**
+   * Starts a Local Service.
+   * <p>
+   * The status of the Local Service is displayed in the center text area of the application.
+   * The URL of the Local Service is added to the list of running services.
+   * 
+   * @param localService Local Service to start.
+   * @param serviceName Name of Local Service. 
+   */
+  private void startLocalService(LocalService localService, String serviceName) {
+    localService.addStatusChangedListener(status -> {
+      centerText.appendText(serviceName + " Status: " + status.getNewStatus().toString() + "\n");
+      if (status.getNewStatus() == LocalServerStatus.STARTED) {
+        Platform.runLater(() -> runningServices.getItems().add(serviceName + " URL ->  " + localService.getUrl()));
+
+      } else if (isServerStopping && status.getNewStatus() == LocalServerStatus.STOPPED) {
+        int servicesStarted = services.size();
+        for (LocalService service : services) {
+          if (service.getStatus() == LocalServerStatus.STOPPED) {
+            servicesStarted--;
+          }
+        }
+        if (servicesStarted == 0) {
+          System.out.println("Service Stopped");
+          stopServer();
+        }
+      }
+    });
+    localService.startAsync();
+  }
+
+  /**
+   * Gets the Local Service that is selected from the list of running services and stops it. 
+   * 
+   * @param event the event that fired this action
+   */
+  @FXML
+  protected void handleStopService(ActionEvent event) {
+
+    int serviceIndex = runningServices.getSelectionModel().getSelectedIndex();
+    services.get(serviceIndex).stopAsync();
+    runningServices.getItems().remove(serviceIndex);
+  }
+
+  /**
+   * Enables the URL and Stop Service button when a service is selected.
+   */
+  @FXML
+  protected void enableServiceButtons() {
+
+    int listIndex = runningServices.getSelectionModel().getSelectedIndex();
+    if (listIndex != -1) {

```

Comment:
it might be useful to create observable properties for service statuses. Then you could just bind the button's disabled property to that status property instead of manually calling `setDisable()`.